src/NUnitFramework/framework/Constraints/Comparers/LineEndingNormalizingStringComparer.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NUnit.Framework.Constraints.Comparers;
+
+/// <summary>
+/// Normalizes different line endings (\r, \n, and \r\n) to \n before comparig strings.
+/// </summary>
+internal sealed class LineEndingNormalizingStringComparer : IEqualityComparer<string>
+{
+    /// <summary>
```

Comment:
Only two of these are used, but I just created singletons for all 6 `StringComparison` values.